### PR TITLE
First data processing functionality

### PR DESCRIPTION
**Pull request by wis-janusz**
Added xcf/data/proc_data.py with public function for general data processing for different modules. 2 private functions: pHpredictor data clean-up and one-hot encoding.
Additionally added notebook with exploratory data analysis and implementation of data processing for pHpredictor.

### CODE REVIEW

File: xcf/data/proc_data.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    @classmethod
+    def pHpred_range(cls):
+        return cls._pHpred_range
+
+    @classmethod
+    def pHpred_seq_range(cls):
+        return cls._pHpred_seq_range  
+
+
+def _pHpred_cleaner(data_raw:pd.DataFrame, pHpred_range:tuple, pHpred_seq_range:tuple) -> pd.DataFrame:
```
Comment (angon1) [inline xcf/data/proc_data.py:28]: Consider grouping bellow functions in class or seperate module.  They have common use case.

File: xcf/data/proc_data.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    return data_proc
+
+def _one_hot_encode_aa(sequence:str, max_len:int, encoder) -> pd.DataFrame:
+        
+    sequence_padded = sequence.ljust(max_len, '#')
+    seq_array = np.array(list(sequence_padded)).reshape(-1,1)
+        
+    return encoder.fit_transform(seq_array).astype('int8')
+
+def process_data(data_raw:pd.DataFrame, *, mode:str = DataProcConfig.pHpred()) -> pd.DataFrame:
```
Comment (angon1) [inline xcf/data/proc_data.py:45]: `mode = DataProcConfig.pHpred()` as argument is bad idea.
default arguments are evaluated only once - when function is defined.  Better idea is to handle it inside your func:
```python 
def process_data(data_raw:pd.DataFrame, *, mode:str = None) -> pd.DataFrame:
    if not mode:
        mode = DataProcConfig.pHpred()
```

File: xcf/data/proc_data.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import numpy as np
```
Comment (angon1) [inline xcf/data/proc_data.py:1]: Please consider adding `__init__.py` in data folder. 
Also you may consider using `__init__.py` to handle package imports. example content:
```python
from .proc_data.py import DataProcConfig
form .proc_data.py import process_data
```

then in some other file in xcf package you may use:
```python
from data import DataProcConfig

# or:
import data
data.DataProcConfig()...
```

**Review by angon1 — APPROVED** [review]

File: xcf/data/proc_data.py (hunk 3 above)
Comment (wis-janusz) [reply xcf/data/proc_data.py:1]: Got it, will change

File: xcf/data/proc_data.py (hunk 2 above)
Comment (wis-janusz) [reply xcf/data/proc_data.py:45]: OK, will be changed

**Pull request closed without merging**